siriuspy/siriuspy/clientarch/pvarch.py

```diff
@@ -3,29 +3,38 @@
 from .client import ClientArchiver as _ClientArchiver
 
 
-class PVArch:
-    """PVArch."""
+class PVDetails:
+    """PV Arch Details."""
 
-    def __init__(self, pvname, client_archiver=None):
+    def __init__(self, pvname, connector=None):
         """."""
         self.pvname = pvname
+        self.connector = connector
         self.is_scalar = None
+        self.is_paused = None
+        self.is_connected = None
         self.nelms = None
         self.units = None
-        self.is_paused = None
         self.host_name = None
-        self.connected = None
         self.avg_bytes_per_event = None
         self.estimated_storage_rate_kb_hour = None
         self.estimated_storage_rate_mb_day = None
         self.estimated_storage_rate_gb_year = None
 
-        self.connector = client_archiver
+    @property
+    def request_url(self):
+        """."""
+        url = self.connector.getPVDetails(self.pvname, get_request_url=True)
```

Comment:
there is a `self.connect()` missing here.

Reply:
done